wordpress-mobile/WordPress-Android
# Detekt - Resolve/Suppress All Baseline Warnings - Long methods warnings_ HomePageSettingsDialog, AuthorUseCase, ClicksUseCase, ReferrersUseCase

Parent: https://github.com/wordpress-mobile/WordPress-Android/issues/17010

This PR resolves/suppresses all complexity related LongMethod warnings for the HomePageSettingsDialog class, AuthorUseCase class, ClicksUseCase class, and ReferrersUseCase class (see docs [here](https://detekt.dev/docs/rules/complexity#longmethod)):

`4` x LongMethod  (Resolve: ad6bb237a2fb36f796156f9bede0bffd9a66a846 + 1cbf13249099f737bde8e23c46b9dff45fb4f1ff + f887ee07a8cb37ed37dc3b7e545ea717ca435e9e + 4e0cee3cf46405215c5b3cc9c98fdab97bc3bb84 + ccbb810665b27e5e88b2f0f66872046d02ff87e1 + 0431cde47a8ea2377cd69e177b31f66833bf6ea8 + 55311df61850b15373180b119e20fe5d5c065ec5 + 9432e4812c388ff9e87833b4e6eed9df350ab938 )

-----

To test:

- There is nothing much to test here.

- Verifying that all the CI checks are successful should be enough (especially the detekt check).
- However, if you really want to be thorough, you could smoke test the WordPress and/or Jetpack apps to verify that everything works as expected on every screen that relates to these changes. Specifically, you could follow the below steps:

   - Go to the menu tab
   - Click on site settings
   - Select HomePage settings
   - Test that you can change between different homepage settings 

To Test `Stats` 
   - Go to the menu tab
   - Click on Stat
   - Click on `Days` tab and confirm that the correct data for `Referrers`, `Clicks`, and `Authors` are displaying 
   - Do same for `weeks`, `months` and `year` tab 

## Regression Notes
1. Potential unintended areas of impact
     can't think of any 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)
     N/A

PR submission checklist:
